StRoot/StEvent/StFwdTrack.h

```diff
@@ -0,0 +1,187 @@
+/***************************************************************************
+ *
+ * $Id: StFwdTrack.h,v 2.1 2021/01/11 20:25:37 ullrich Exp $
+ *
+ * Author: jdb, Feb 2022
+ ***************************************************************************
+ *
+ * Description: StFwdTrack stores the Forward tracks built from Fst and Ftt
+ *
+ ***************************************************************************
+ *
+ * $Log: StFwdTrack.h,v $
+ * Revision 2.1  2021/01/11 20:25:37  ullrich
+ * Initial Revision
+ *
+ **************************************************************************/
+#ifndef StFwdTrack_hh
+#define StFwdTrack_hh
+
+#include "Stiostream.h"
+#include "StObject.h"
+#include <vector>
+#include "StThreeVectorD.hh"
+#include "StContainers.h"
+
+class StFcsCluster;
+
+
+struct StFwdTrackProjection {
+    StFwdTrackProjection() {}
+    StFwdTrackProjection ( const StFwdTrackProjection & other) {
+        mXYZ = other.mXYZ;
+        mMom = other.mMom;
+        mDetId = other.mDetId;
+        memcpy( mCov, other.mCov, sizeof( mCov ) );
+    }
+    StFwdTrackProjection(   unsigned short detId, 
+                            StThreeVectorD xyz, 
+                            StThreeVectorD mom, 
+                            float c[9] ) {
+        set( detId, xyz, mom, c );
+    }
+
+    void set(   unsigned short detId, 
+                StThreeVectorD xyz, 
+                StThreeVectorD mom, 
+                float c[9]) {
+        mDetId = detId;
+        mXYZ = xyz;
+		mMom = mom;
+        memcpy( mCov, c, sizeof(mCov) ); 
+    }
+    void set(   StFwdTrackProjection &other ){
+        mDetId = other.mDetId;
+        mXYZ   = other.mXYZ;
+        mMom   = other.mMom;
+        memcpy( mCov, other.mCov, sizeof(mCov) ); 
+    }
+    StThreeVectorD mXYZ;
+	StThreeVectorD mMom;
+    unsigned char mDetId;
+    float mCov[9];
+
+    float dx(){
+        return sqrt( mCov[0] );
+    }
+    float dy(){
+        return sqrt( mCov[4] );
+    }
+    float dz(){
+        return sqrt( mCov[8] );
+    }
+
+    enum {
+        PrimaryVertex = 0,
+        FST = 1,
+        FTT = 2,
+        EPD = 3,
+        ECAL = 4, 
+        HCAL = 5
+    } DetectorId;
```

Comment:
We discussed this in Fwd meeting, but there is no identifier for primary vertex, also these act as indexes in an array, so it is not practical to use existing unless a map is used instead.

Reply:
OK I changed some things around and implemented this using the StEvent enum instead of a new one. See [50b0903](https://github.com/star-bnl/star-sw/pull/492/commits/50b0903ff883099b68962cd37f2e8e645a4fd6c6)